flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -592,19 +605,18 @@ class TaskManager(
             log.warn(s"Ignoring 'AcknowledgeRegistration' message from ${jobManager.path} , " +
               s"because the TaskManager is already registered at ${currentJobManager.orNull}")
           }
-        }
-        else {
+        } else {
           // not yet connected, so let's associate with that JobManager
           try {
             associateWithJobManager(jobManager, id, blobPort)
           } catch {
             case t: Throwable =>
               killTaskManagerFatal(
-                "Unable to start TaskManager components after registering at JobManager", t)
+                "Unable to start TaskManager components and associate with the JobManager ", t)
           }
         }
 
-      // we are already registered at that specific JobManager - duplicate answer, rare cases
+      // we are already registered at this ResourceManager - duplicate answer, rare cases
       case AlreadyRegistered(id, blobPort) =>
         val jobManager = sender()
```

Comment:
This is an artifact of refactoring the original approach that I followed where the TaskManager would register at the ResourceManager. Reverting.